service/src/main/java/tds/session/repositories/ExternRepository.java

```diff
@@ -0,0 +1,14 @@
+package tds.session.repositories;
+
+import java.util.Optional;
+
+import tds.session.Extern;
+
+public interface ExternRepository {
+    /**
+     * Fetches the extern by given client name
+     * @param clientName client name for extern
+     * @return optional containing {@link tds.session.Extern Extern} otherwise empty Optional
+     */
+    Optional<Extern> getExternByClientName(String clientName);
```

Comment:
It was decided that for now we'll let clients handle the error case if externs are not available.  So this will still return a 404 and the caller can throw a 500 if warranted.  In the near term that will always be the case.